django_mongodb/base.py

```diff
@@ -0,0 +1,126 @@
+        "DateTimeField": "date",
+        "DecimalField": "decimal",
+        "DurationField": "long",
+        "FileField": "string",
+        "FilePathField": "string",
+        "FloatField": "double",
+        "IntegerField": "int",
+        "BigIntegerField": "long",
+        "GenericIPAddressField": "string",
+        "NullBooleanField": "bool",
+        "OneToOneField": "int",
+        "PositiveIntegerField": "long",
+        "PositiveSmallIntegerField": "int",
+        "SlugField": "string",
+        "SmallIntegerField": "int",
+        "TextField": "string",
+        "TimeField": "date",
+        "UUIDField": "string",
+    }
+
+    vendor = "mongodb"
+    Database = Database
+    SchemaEditorClass = DatabaseSchemaEditor
+    client_class = DatabaseClient
+    creation_class = DatabaseCreation
+    features_class = DatabaseFeatures
+    introspection_class = DatabaseIntrospection
+    ops_class = DatabaseOperations
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.connected = False
+        del self.connection
+
+    def get_collection(self, name, **kwargs):
+        return Collection(self.database, name, **kwargs)
+
+    def __getattr__(self, attr):
+        if attr in ["connection", "database"]:
+            assert not self.connected
+            self._connect()
+            return getattr(self, attr)
+        raise AttributeError(attr)
+
+    def _connect(self):
+        settings_dict = self.settings_dict
+
+        options = settings_dict["OPTIONS"]
+        self.operation_flags = options.pop("OPERATIONS", {})
+        if not any(k in ["save", "delete", "update"] for k in self.operation_flags):
+            # Flags apply to all operations.
+            flags = self.operation_flags
+            self.operation_flags = {"save": flags, "delete": flags, "update": flags}
+
+        conn_options = {
+            "host": settings_dict["HOST"] or None,
+            "port": int(settings_dict["PORT"] or 27017),
+            "document_class": dict,
+            "tz_aware": False,
+        }
+        conn_options.update(options)
+
+        self.connection = MongoClient(**conn_options)
+        db_name = settings_dict["NAME"]
+        if db_name:
+            self.database = self.connection[db_name]
+
+        user = settings_dict["USER"]
+        password = settings_dict["PASSWORD"]
+        if user and password and not self.database.authenticate(user, password):
+            raise ImproperlyConfigured("Invalid username or password.")
+
+        self.connected = True
+        connection_created.send(sender=self.__class__, connection=self)
+
+    def _reconnect(self):
+        if self.connected:
+            del self.connection
+            del self.database
+            self.connected = False
+        self._connect()
+
+    def _commit(self):
+        pass
+
+    def _rollback(self):
+        pass
+
+    def close(self):
+        pass
+
+    def cursor(self):
+        return Cursor()
```

Comment:
This was a quick hack to prevent the need to override some Django internals where a "nodb" cursor is created but (for MongoDB purposes) not used. It may or may not stay around long-term.

Reply:
I'm trying to find more information on cursors in django, any good documentation reference?

Reply:
It's up to the database driver (psycopg, mysqlclient, etc.) to implement, but the interface is described in [PEP 249](https://peps.python.org/pep-0249/#cursor-objects).